fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -406,22 +407,23 @@ public FedoraResource getTimeMap() {
         }
 
         try {
-            final Node timeMapNode;
-            if (isMemento()) {
-                timeMapNode = node.getParent();
+            if (isOriginalResource()) {
+                return findOrCreateTimeMap();
+            } else if (isMemento()) {
+                return Optional.of(node.getParent()).map(nodeConverter::convert).orElse(null);
+
             } else {
-                timeMapNode = node.getNode(LDPCV_TIME_MAP);
+                return null;
```

Comment:
I wonder if we should throw an exception here, since it would be requesting the timemap of something that can't have a timemap?